src/evaluation_system/api/plugin_manager.py

```diff
@@ -1041,18 +1041,14 @@ def get_history(
     QuerySet[History]
         Results from the database query.
     """
-    user = user or User()
-
-    if plugin_name is not None:
-        plugin_name = plugin_name.lower()
-
-    return user.getUserDB().getHistory(
-        plugin_name,
-        limit,
+    user_ = user or User()
+    return user_.getUserDB().getHistory(
+        tool_name=plugin_name.lower() if plugin_name else None,
+        limit=limit,
         since=since,
         until=until,
         entry_ids=entry_ids,
-        uid=user.getName(),
+        uid=user.getName() if user else None,
```

Comment:
So you either want all user or just the current user.

Wouldn't it be better to add a new user_name argument to this function that uses the uid argument. Something like this:

```python
def get_history(                                                                                                                                                           
      plugin_name: Optional[str] = None,                                                                                                                                     
      limit: int = -1,                                                                                                                                                       
      since: Optional[str] = None,                                                                                                                                           
      until: Optional[str] = None,                                                                                                                                           
      entry_ids: Optional[list[int]] = None,                                                                                                                                 
      user: Optional[User] = None,   
      user_name: Optional[str] = None,                                                                                                                                        
  ) -> QuerySet[History]:                                                                                                                                                    
      """Returns the history from the given user.                                                                                                                            
                                                                                                                                                                             
      This is just a wrapper for the defined db interface accessed via the                                                                                                   
      user object. See :class:`evaluation_system.model.db.UserDB.getHistory`                                                                                                 
      for more information on this interface.                                                                                                                                
                                                                                                                                                                             
      Parameters                                                                                                                                                             
      ----------                                                                                                                                                             
      plugin_name                                                                                                                                                            
          Name of plugin to get the history for                                                                                                                              
      limit                                                                                                                                                                  
          Limits on number of results to get. -1 means no limit                                                                                                              
      since                                                                                                                                                                  
          Only get results after since.                                                                                                                                      
      until                                                                                                                                                                  
          Only get results earlier than until.                                                                                                                               
      entry_ids                                                                                                                                                              
          Result entry IDs to filter for.                                                                                                                                    
      user                                                                                                                                                                   
          User object to query results for
       user_name
          User to get plugins results for                                                                                                                       
                                                                                                                                                                             
      Returns                                                                                                                                                                
      -------                                                                                                                                                                
      QuerySet[History]                                                                                                                                                      
          Results from the database query.                                                                                                                                   
      """                                                                                                                                                                    
      user = user or User()                                                                                                                                                  
      user_name = user_name or user.getName()
      if user_name in ["*", "all"]:
         user_name = None                                                                                                                                                                       
      if plugin_name is not None:                                                                                                                                            
          plugin_name = plugin_name.lower()                                                                                                                                  
                                                                                                                                                                             
      return user.getUserDB().getHistory(                                                                                                                                    
          plugin_name,
           limit,                                                                                                                                                             
          since=since,                                                                                                                                                       
          until=until,                                                                                                                                                       
          entry_ids=entry_ids,                                                                                                                                               
          uid=user_name,                                                                                                                                                
      )
```

Reply:
I put it like this